sdk/Pulumi.Automation/WorkspaceStack.cs

```diff
@@ -489,6 +489,23 @@ void OnPreviewEvent(EngineEvent @event)
                 if (options.ExpectNoChanges is true)
                     args.Add("--expect-no-changes");
 
+                if (options.SkipPendingCreates is true)
+                    args.Add("--skip-pending-creates");
+
+                if (options.ClearPendingCreates is true)
+                    args.Add("--clear-pending-creates");
+
+                if (options.ImportPendingCreates?.Any() == true)
+                {
+                    foreach (var item in options.ImportPendingCreates)
+                    {
+                        args.Add("--import-pending-creates");
```

Comment:
Ick I did not realize how grim this was to set on the command line. Very good reason to support this in automation api.